vpr/src/place/noc_place_utils.cpp

```diff
@@ -61,64 +79,108 @@ void reinitialize_noc_routing(const t_noc_opts& noc_opts, t_placer_costs& costs)
     initial_noc_routing();
 
     // Initialize traffic_flow_costs
-    costs.noc_aggregate_bandwidth_cost = comp_noc_aggregate_bandwidth_cost();
-    costs.noc_latency_cost = comp_noc_latency_cost(noc_opts);
+    costs.noc_cost_terms.aggregate_bandwidth = comp_noc_aggregate_bandwidth_cost();
+    std::tie(costs.noc_cost_terms.latency, costs.noc_cost_terms.latency_overrun) = comp_noc_latency_cost();
+    costs.noc_cost_terms.congestion = comp_noc_congestion_cost();
 }
 
-void find_affected_noc_routers_and_update_noc_costs(const t_pl_blocks_to_be_moved& blocks_affected, double& noc_aggregate_bandwidth_delta_c, double& noc_latency_delta_c, const t_noc_opts& noc_opts) {
-    // provides the positions where the affected blocks have moved to
-    auto& place_ctx = g_vpr_ctx.placement();
+void find_affected_noc_routers_and_update_noc_costs(const t_pl_blocks_to_be_moved& blocks_affected,
+                                                    NocCostTerms& delta_c) {
+    VTR_ASSERT_SAFE(delta_c.aggregate_bandwidth == 0.);
```

Comment:
Add comment: For speed, we expect the delta_c structure to be passed in. It must start out with 0 costs for correctness.